gammapy/data/metadata.py

```diff
@@ -60,46 +59,23 @@ class ObservationMetaData(MetaData):
         Additional optional metadata.
     """
 
-    _tag = "observation"
-    obs_info: Optional[ObsInfoMetaData]
-    pointing: Optional[PointingInfoMetaData]
-    target: Optional[TargetMetaData]
-    location: Optional[Union[EarthLocation, str]]
+    _tag: ClassVar[Literal["observation"]] = "observation"
```

Comment:
This does two things: 
- `_tag` is currently not part of the model definition, because by convention Pydantic treats attributes starting with underscore as private attributes. This is also why the `ClassVar` is needed. The tag is assigned to the class not the instance.
- The Literal only allows exactly this value. However here it has not a real function, except for documentation. 

I would actually propose to make the tag part of the model definition and use the `Literal`. This will fix the value to the value defined by the Literal.